src/commands/basemaps-github/github.ts

```diff
@@ -0,0 +1,116 @@
+import { Env, LogType } from '@basemaps/shared';
+import { execFileSync } from 'child_process';
+import { Octokit } from '@octokit/core';
+import { Api } from '@octokit/plugin-rest-endpoint-methods/dist-types/types.js';
+import { restEndpointMethods } from '@octokit/plugin-rest-endpoint-methods';
+
+export class Github {
+  repo: string;
+  org: string;
+  logger: LogType;
+  repoName: string;
+  octokit: Api;
+
+  constructor(repo: string, org: string, repoName: string, logger: LogType) {
+    this.repo = repo;
+    this.logger = logger;
```

Comment:
ideally  the logger should be passed into each operations than at the top level, this is mostly so this class could be accessed concurrently if needed.

Reply:
Will fix next PR